mesonbuild/modules/gnome.py

```diff
@@ -45,7 +45,11 @@ def compile_resources(self, state, args, kwargs):
         if not isinstance(source_dirs, list):
             source_dirs = [source_dirs]
 
-        ifile = args[1]
+        try:
+            ifile = args[1]
+        except IndexError:
+            raise mesonlib.MesonException('Missing an argument')
```

Comment:
This can just be `MesonException` because we import that already.

Also, would be nice to have a descriptive error message. :smile: Perhaps something like: 

"The first argument must be the name of the resource and the second must be the XML file"